### ISSUE TITLE
[Crash] Conflict between TerraFirmaCraft and PrimalCore.

### ISSUE THREAD

**Issue by UnlimatedStone9** [An-Sar/PrimalCore#350]
https://paste.dimdev.org/firuwikipo.mccrash

**Comment by UnlimatedStone9** [comment]
Both PrimalCore version 1.12.2-0.6.105 and PrimalCore version 1.12.2-0.6.104 appear to have this issue

**Comment by UnlimatedStone9** [comment]
I came to the conclusions it was a fault in PrimalCore using a process of elimination, removing individual mods from the pack.
From testing it appears PrimalCore and one of the most recent builds of TerraFirmaCraft appear to be conflicting, unusual as both mods worked fine a few versions back.
Neither TerraFirmaCraft nor PrimalCore can be present at the same time but both launch fine on their own, I'll report this to the TFC team too!

**Comment by An-Sar** [comment]
thanks for narrowing it down, i am going to get caught up on the 1.12 issues soon. i will try to add terrafirma to dev and see what happens.

**Comment by UnlimatedStone9** [comment]
I'll happily link you to the Jenkins download for TFC and the GitHub
https://jenkins.dries007.net/TerraFirmaCraft
https://github.com/TerraFirmaCraft/TerraFirmaCraft

**Comment by alcatrazEscapee** [comment]
Do you by change registry replace `minecraft:torch` and keep the instance around, or use an object holder with a custom subclass?

**Comment by UnlimatedStone9** [comment]
By the looks of it, disabling the torch replacement in PrimalCores config file seems to eliminate the crashing and results in a fully successful game launch, although I would still like to see some compatibility fix so as to allow me to use PrimalCores' unique torch mechanics, i.e the smoke build up and such alongside TerraFirmaCraft being installed.

I believe this confirms torches seem to be the issue.

**Comment by An-Sar** [comment]
@alcatrazEscapee yes the torch override uses an ObjectHolder, there is a config to disable the feature as @UnlimatedStone9 just mentioned. 

i wonder if TerraFirma is also overriding the minecraft:torch

**Comment by alcatrazEscapee** [comment]
Yes, we do. I've checked that there aren't any references to the old registered block or item that we're holding around, and by the looks of it I can't see any. I can try and look into it and see exactly what's causing a problem - by the looks of it there's an additional block / item lying around that is not getting registry replaced correctly.

Edit: Something is funky here. I'm ~~90%~~ (it does) sure that TFC's replacement runs after, in which case this `VanillaTorchItem` is primal core holding around their instance somehow
![weird_registry_override](https://user-images.githubusercontent.com/35673674/64290799-c745cc00-cf34-11e9-81ec-317806f0e003.PNG)
